lib/ui/recommendation.js

```diff
@@ -123,21 +123,25 @@ Recommendation.prototype = {
     this.events.publish('recommendation-mousemove', evt);
   },
   show: function(data) {
+    let payload = data.payload;
+    let ResultClass = data.resultClass;
+
     if (this.timeout) {
       this.win.clearTimeout(this.timeout);
       this.timeout = null;
     }
 
     // Don't re-render the recommendation if the content hasn't changed since
     // the previous recommendation.
-    if (this.prev && this.el && this.el.firstChild && data.result.url === this.prev.result.url) {
+    if (this.prev && this.el && this.el.firstChild &&
```

Comment:
Rather than check for `this.el` here and at line 149 below, just check once, at the very top of the function (before the timeout check).